app/src/main/java/com/example/drawingApp/ColorPickerUtility.kt

```diff
@@ -0,0 +1,222 @@
+import kotlin.math.max
+import kotlin.math.min
+
+object ColorPickerUtility {
+
+    @SuppressLint("ClickableViewAccessibility")
+    fun colorPickerSheet(
+        context: Context,
+        /**
+         * Called whenever the hue is updated via text or slider.
+         * Use it to set wherever you are storing your hue
+         * and then update your color
+         */
+        onHueUpdate: (Float) -> Unit,
+        /**
+         * Called whenever the Saturation or Value is updated via the gradient square
+         * or via text. use it to set wherever you are storing your saturation AND value
+         * update your color afterward
+         */
+        onSatOrValUpdate: (Float, Float) -> Unit,
+        /**
+         * Called whenever text needs to be updated, meaning on any change, hsv, rgb, or text.
+         * Used to give the information of hsv and previous color.
+         * if there was no previous color pass it black. This is used to help
+         * populate the sheets text views and update the slider and gradient indicators as well as set the previous color box.
+         */
+        onColorTextForceUpdate: () -> DrawingViewModel.ColorPack,
+        /**
+         * Called whenever one of the RGB texts is changed by focus by a user.
+         * it passes a ColorInt as the first parameter, used to decide what value to set.
+         * the second int is the actual value you will set your R,G,or B value to.
+         * Color.RED corresponds to red,Color.GREEN to green, BLUE to blue
+         */
+        onRGBTextChange: (Int, Int) -> Unit,
+        /**
+         * called whenever the dialog is dismissed or submitted
+         * on actual submission the color will not be null so take it in and set it wherever you need the selected color and set the hsv and rgb fields to defaults.
+         * on null don't do anything with it, don't set rgb and hsv to defaults.
+         */
+        onSubmission: (Color?) -> Unit
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val inflater = LayoutInflater.from(context)
+        val view = inflater.inflate(R.layout.color_picker_bottom_sheet, null)
+        val submit = view.findViewById<TextView>(R.id.dialogSubmit)
+        val cancel = view.findViewById<TextView>(R.id.dialogCancel)
+        val hView = view.findViewById<ForceEditText>(R.id.hslH)
+        val sView = view.findViewById<ForceEditText>(R.id.hslS)
+        val vView = view.findViewById<ForceEditText>(R.id.hslL)
+        val rView = view.findViewById<ForceEditText>(R.id.rgbR)
+        val gView = view.findViewById<ForceEditText>(R.id.rgbG)
+        val bView = view.findViewById<ForceEditText>(R.id.rgbB)
+        val chView = view.findViewById<ForceEditText>(R.id.colorHex)
+        val slider = view.findViewById<ColorSlider>(R.id.slider)
+        val square = view.findViewById<GradientSquare>(R.id.gradientSquare)
+        val activeBox = view.findViewById<TextView>(R.id.colorOne)
```

Comment:
Very confused on what active box is, this the current color? looks like it to me

Reply:
it's the current color thats actively shown and edited by the user.

Reply:
adjusted to activeColorBox